contracts/hydro/src/contract.rs

```diff
@@ -1700,8 +1708,60 @@ pub fn query_all_user_lockups(
         })
         .collect();
 
+    // enrich lockups with some info per tranche
+    let lockups_with_per_tranche_info: Vec<LockupWithPerTrancheInfo> = locks_with_power
+        .iter()
+        .map(|lock| {
+            // iterate all tranches
+            let tranche_infos = tranche_ids
+                .iter()
+                .map(|tranche_id| {
+                    // add which proposal the lock voted for
+                    let voted_for_proposal: Option<u64> = VOTE_MAP
+                        .may_load(
+                            deps.storage,
+                            (
+                                (current_round_id, *tranche_id),
+                                converted_addr.clone(),
+                                lock.lock_entry.lock_id,
+                            ),
+                        )
+                        .unwrap_or(None)
+                        .map(|vote| vote.prop_id);
+
+                    let next_round_voting_allowed: u64 = if voted_for_proposal.is_some() {
+                        // if the proposal voted in this round, we ignore the VOTING_ALLOWED_ROUND map,
+                        // since it just contains the future information on when the lockup will be able to vote again
+                        // if it doesn't change the vote, but it can vote in the current round either way
+                        current_round_id
+                    } else {
+                        // if the lockup has not voted in this round, VOTING_ALLOWED_ROUND does contain
+                        // current information on whether the votup can vote right now or not
+                        VOTING_ALLOWED_ROUND
+                            .may_load(deps.storage, (*tranche_id, lock.lock_entry.lock_id))
+                            .unwrap_or_default()
+                            .unwrap_or(0)
+                    };
+
+                    // return the info for this tranche
+                    PerTrancheLockupInfo {
+                        tranche_id: *tranche_id,
+                        next_round_lockup_can_vote: next_round_voting_allowed,
+                        current_voted_on_proposal: voted_for_proposal,
```

Comment:
This is a bit misleading since it will only give info for which proposal the lockup voted for in the current round (if it voted). Maybe we are OK with that? Not sure is it worth to search for which proposal it voted in latest round before the current one.

Reply:
Yeah, this is just in the current round. I will add a comment. Doing this in general seems annoying with our state structure right now.

I will leave it as is for now, but it's one thing we might want to add later on